pkg/extensions/sync/service.go

```diff
@@ -243,7 +243,7 @@ func (service *BaseService) SyncImage(repo, reference string) error {
 	service.log.Info().Str("remote", remoteURL).Str("repo", repo).Str("reference", reference).
 		Msg("sync: syncing image")
 
-	manifestDigest, err := service.syncTag(repo, remoteRepo, reference)
+	manifestDigest, err := service.syncTag(context.Background(), repo, remoteRepo, reference)
```

Comment:
Need to get that context here as well

Reply:
this is for on demand, we don't want to cancel, it will run in background and if a new client request the same image it will wait on image channel to be downloaded.

Reply:
But does zot wait for this then?

Reply:
The main idea, being from a top-level context, we either force cancel or we wait for done. We have to pick one.

Reply:
And ideally all foreground/background tasks are accounted for.

Reply:
yes, my bad, updated, thank you!